.github/workflows/tests.yml

```diff
@@ -4,78 +4,126 @@ on: [pull_request, push]
     - name: conda env update
       if: steps.cache.outputs.cache-hit != 'true'
       run: |
         source '/usr/share/miniconda/etc/profile.d/conda.sh'
-        conda env update -q --file environment.yml --name improver
-        conda list --export
-    - name: conda activate
-      run:  echo "/usr/share/miniconda/envs/improver/bin" >>$GITHUB_PATH
+        conda env update -q --file envs/environment_${{ matrix.env }}.yml --name improver_${{ matrix.env }}
+    - name: conda info
+      run: |
+        source '/usr/share/miniconda/etc/profile.d/conda.sh'
+        conda activate improver_${{ matrix.env }}
+        conda info
+        conda list
     - name: isort
-      run:  isort --check-only --recursive .
+      run: |
+        source '/usr/share/miniconda/etc/profile.d/conda.sh'
+        conda activate improver_${{ matrix.env }}
+        isort --check-only --recursive .
     - name: black
-      run:  black --check .
+      run: |
+        source '/usr/share/miniconda/etc/profile.d/conda.sh'
+        conda activate improver_${{ matrix.env }}
+        black --check .
     - name: pylint
-      run: pylint -j 0 -E --rcfile=etc/pylintrc improver improver_tests
+      run: |
+        source '/usr/share/miniconda/etc/profile.d/conda.sh'
+        conda activate improver_${{ matrix.env }}
+        pylint -j 0 -E --rcfile=etc/pylintrc improver improver_tests
   Safety-Bandit:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        env: [py36_iris22, py37_iris24]
     steps:
     - uses: actions/checkout@v2
     - uses: actions/cache@v1
       id: cache
+      env:
+        # Increase this value to reset cache
+        CACHE_NUMBER: 1
       with:
-        path: /usr/share/miniconda/envs/improver
-        key: ${{ runner.os }}-conda-improver-${{ hashFiles('**/environment.yml') }}
-        restore-keys: ${{ runner.OS }}-conda-improver-
+        path: /usr/share/miniconda/envs/improver_${{ matrix.env }}
+        key: ${{ format('{0}-conda-improver-{1}-{2}-{3}', runner.os, env.CACHE_NUMBER, matrix.env, hashFiles(format('envs/environment_{0}.yml', matrix.env))) }}
     - name: conda env update
       if: steps.cache.outputs.cache-hit != 'true'
       run: |
         source '/usr/share/miniconda/etc/profile.d/conda.sh'
-        conda env update -q --file environment.yml --name improver
-        conda list --export
-    - name: conda activate
-      run:  echo "/usr/share/miniconda/envs/improver/bin" >>$GITHUB_PATH
+        conda env update -q --file envs/environment_${{ matrix.env }}.yml --name improver_${{ matrix.env }}
+    - name: conda info
+      run: |
+        source '/usr/share/miniconda/etc/profile.d/conda.sh'
+        conda activate improver_${{ matrix.env }}
+        conda info
+        conda list
     - name: safety
-      run:  safety check || true
+      run: |
+        source '/usr/share/miniconda/etc/profile.d/conda.sh'
+        conda activate improver_${{ matrix.env }}
+        safety check || true
```

Comment:
In stage, we allow bandit and safety to return failure, but the repo has been configured so that these failures don't block merging. (See https://github.com/MetOffice/stage/pull/35#pullrequestreview-553432072). Maybe we should do same here to be consistent? CC @owena11

Reply:
Thanks for the review @btrotta-bom. This change was originally implemented in #1385. I'm happy to make that change but maybe it could be in a separate PR. This one is already getting large.